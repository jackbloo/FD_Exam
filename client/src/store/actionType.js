export const LOGIN = 'LOGIN'
export const FETCHPASSWORD = 'FETCHPASSWORD'
export const LOGOUT = 'LOGOUT'
export const FETCHDETAIL = 'FETCH DETAIL'
export const SETEDIT = 'SETEDIT'
export const SHOW = 'SHOW'
export const UNSHOW = 'UNSHOW'
export const LOAD = 'LOAD'
export const UNLOAD = 'UNLOAD'
export const DELETE = 'DELETE'