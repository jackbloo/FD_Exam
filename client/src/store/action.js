import {LOGIN, LOGOUT, FETCHPASSWORD,SHOW,UNSHOW,LOAD,UNLOAD, FETCHDETAIL} from '../store/actionType'


export const show = () => ({type:SHOW})
export const unshow = () => ({type:UNSHOW})
export const login = () => ({type:LOGIN})
export const logout = () => ({type:LOGOUT})
export const load = () => ({type:LOAD})
export const unload = () => ({type: UNLOAD})
export const fetchPassword = (data) =>  ({type:FETCHPASSWORD, data})


// export const deleteOnePass = (id) => dispatch => {
//     dispatch({type:LOAD})
//     pm
//     .doc(id)
//     .delete()
//     .then(function() {
//         dispatch({type:UNLOAD})
//         toast.success('Your Password is deleted')
//     }).catch(function(error) {
//         toast.error(error)
//     });
// }
