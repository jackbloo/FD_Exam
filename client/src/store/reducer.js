import {GETDATA} from './actionType'

const initialState = {
    editorChoice:[],
    latestArticles:[],
    latestReview:[],
    navbar: [{id:'skincare'},{id:'make up'},{id:'body'},{id:'hair'},{id:'fragrance'},{id:'nails'},{id:'tools'},{id:'brands'}],
    linkData:[{id: 'About Us'}, {id: 'Terms & Conditions'}, {id: 'Awards'},{id: 'Feedback'},{id: 'Privacy Policy'}, {id: 'Newsletter'}, {id: 'Content'},{id: 'Help'}]
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case GETDATA:
            const data = action.data
            return {
                ...state,
                editorChoice: [...data["editor's choice"]],
                latestArticles:[...data["latest articles"]],
                latestReview:[...data["latest review"]]
            }
        default:
            return state     
    }
 
}

export default reducer