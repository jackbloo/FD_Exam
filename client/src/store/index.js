import { configureStore } from '@reduxjs/toolkit'
import reducer from '../store/reducer'

const store = configureStore({reducer: reducer})

export default store