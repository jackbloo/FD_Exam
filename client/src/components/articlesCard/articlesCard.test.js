import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ArticlesCard from './articlesCard'
import { Provider} from 'react-redux'
import store from '../../store'


describe('Test ArticlesCard', () => {
    const data = {
        image:'131231.png',
        title: 'Test',
        author: 'Test-1',
        published_at: '2 hours ago'
    }
    it('Should Render Card match snapshot', ()=>{

     render(
            <Provider store={store}>
                    <ArticlesCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('article-card')
        expect(card).toMatchSnapshot()
    })


    it('Should Render Card without crashing', ()=>{

     render(
            <Provider store={store}>
                    <ArticlesCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('article-card')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
        expect(card).toHaveTextContent("TestTest-1 |2 hours ago")
        expect(card).not.toHaveTextContent('Facebook.com')
    })

    
    it('Should Render article image', ()=>{

     render(
            <Provider store={store}>
                    <ArticlesCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('article-image')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render article title', ()=>{

     render(
            <Provider store={store}>
                    <ArticlesCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('article-title')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render article username', ()=>{

     render(
            <Provider store={store}>
                    <ArticlesCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('article-username')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render article published', ()=>{

     render(
            <Provider store={store}>
                    <ArticlesCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('article-username')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })


})

