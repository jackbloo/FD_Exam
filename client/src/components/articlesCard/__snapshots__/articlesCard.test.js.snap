// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ArticlesCard Should Render Card match snapshot 1`] = `
<div
  class="sc-bdvvtL jBONVe"
  data-testid="article-card"
>
  <img
    class="sc-gsDKAQ bvZcFw"
    data-testid="article-image"
    src="131231.png"
  />
  <div
    class="sc-dkPtRN jhxwTc"
  >
    <div
      class="sc-hKwDye kJoaCL"
      data-testid="article-title"
    >
      Test
    </div>
    <div
      class="sc-eCImPb fccjuw"
    >
      <div
        class="sc-jRQBWg enHBJw"
        data-testid="article-username"
      >
        Test-1
         
      </div>
      <div
        class="sc-jRQBWg ducLRT"
      >
        |
      </div>
      <div
        class="sc-jRQBWg ducLRT"
        data-testid="article-published"
      >
        2 hours ago
      </div>
    </div>
  </div>
</div>
`;
